README.md

```diff
@@ -1206,6 +1206,158 @@ When defining a new *ramp*, the colors can be an RGB array (`[number, number, nu
 
 Many methods are available on color ramps, such as getting the `<canvas>` element of it, rescaling it, flipping it or [resampling it in a non-linear way](colorramp.md). Read more on [our reference page](https://docs.maptiler.com/sdk-js/api/color-ramp/) and have a look at our [examples](https://docs.maptiler.com/sdk-js/examples/?q=colorramp) to see how they work.
 
+# Camera routes and animations
+
+The SDK comes with several classes to help with animations, particularly route animations.
+
+See `demos/01-animated-routes.html` for examples.
+See `demos/02-maptiler-animation.html` for examples.
+
+#### 🧩 `MaptilerAnimation`
+
+MaptilerAnimation is a utility class for smoothly animating between keyframes using custom easing and playback control. It supports event-based hooks for frame updates and completion, and works well within rendering loops or UI transitions.
+
+##### 🚀 Usage
+
+```ts
+// linearly animated between values
+const animation = new MaptilerAnimation({
+  keyframes: [
+    // `props` are interpolated across the duration
+    { delta: 0, props: { lon: -7.445, } },
+    // `userData` can hold any type of custom data to pass with the keyframe
+    { delta: 0.5, userData: { mydata: "whoa!" } },
+    { delta: 1, props: { lon: -7.455 } }
```

Comment:
Intuitively I'd say that there is no animation since all keyframes are the same (apart from userData). `lon` should be different values in the two keyframes, shouldn't it?